pkg/resources/validation/namespace/namespace.go

```diff
@@ -38,17 +40,63 @@ func (v *Validator) GVR() schema.GroupVersionResource {
 }
 
 // Operations returns list of operations handled by this validator.
-func (v *Validator) Operations() []admissionregistrationv1.OperationType {
-	return []admissionregistrationv1.OperationType{
-		admissionregistrationv1.Update,
-		admissionregistrationv1.Create,
+func (v *Validator) Operations() []admv1.OperationType {
+	return []admv1.OperationType{
+		admv1.Update,
+		admv1.Create,
 	}
 }
 
 // ValidatingWebhook returns the ValidatingWebhook used for this CRD.
-func (v *Validator) ValidatingWebhook(clientConfig admissionregistrationv1.WebhookClientConfig) *admissionregistrationv1.ValidatingWebhook {
+func (v *Validator) ValidatingWebhook(clientConfig admv1.WebhookClientConfig) []admv1.ValidatingWebhook {
 	// Note that namespaces are actually CLUSTER scoped
-	return admission.NewDefaultValidatingWebhook(v, clientConfig, admissionregistrationv1.ClusterScope)
+
+	// standardWebhook validates all operations other than create on all namespaces.
+	standardWebhook := admission.NewDefaultValidatingWebhook(v, clientConfig, admv1.ClusterScope)
+	// the default configuration created will set the operations for each rule equal to the list returned by (*Validator).Operations()
+	// we want to remove the create operation from these list if they exist
+	for i, rule := range standardWebhook.Rules {
+		idx := slices.Index(rule.Operations, admv1.Create)
+		if idx != -1 {
+			standardWebhook.Rules[i].Operations = slices.Delete(rule.Operations, idx, idx+1)
+		}
+	}
+
+	// default configuration for all create operations except those belonging to the kube-system namespace
+	createWebhook := admission.NewDefaultValidatingWebhook(v, clientConfig, admv1.ClusterScope)
+	createWebhook.Name = "rancher.cattle.io.namespace.create.non.kubesystem"
```

Comment:
This value seems to me like something that should only partially be defined in this package. The `rancher.cattle.io` part of it isn't determined by a specific validator/mutator wide. I think it might be better to have a function (something like `admission.SetWebhookName(resourceType string, suffix string)` which takes in a specified suffix and crafts the webhook's name, so we don't end up with different re-definitions of the validator's name.